relayer/relays/beefy/polkadot-listener.go

```diff
@@ -106,15 +113,12 @@ func (li *PolkadotListener) scanCommitments(
 				case <-ctx.Done():
 					return ctx.Err()
 				case requests <- task:
+					logEntry.Info("New commitment with handover added to channel")
+					currentValidatorSet++
 				}
-			} else if result.SignedCommitment.Commitment.ValidatorSetID == currentValidatorSet {
+			} else if result.SignedCommitment.Commitment.ValidatorSetID == currentValidatorSet && result.SignedCommitment.Commitment.ValidatorSetID == uint64(result.MMRProof.Leaf.BeefyNextAuthoritySet.ID) {
```

Comment:
If we have a handover at `currentValidatorSet == setNumber`, we have
```
currentValidatorSet == setNumber
Commitment.ValidatorSetID == setNumber + 1
BeefyNextAuthoritySet.ID == setNumber + 2
```

Then `currentValidatorSet` increments and we have a non-handover case where all of the ids are expected to be equal:
```
currentValidatorSet == setNumber + 1
Commitment.ValidatorSetID == setNumber + 1
BeefyNextAuthoritySet.ID == setNumber + 1
```

So `BeefyNextAuthoritySet.ID` decreases after a handover. I expected this instead:
```
currentValidatorSet == setNumber + 1
Commitment.ValidatorSetID == setNumber + 1
BeefyNextAuthoritySet.ID == setNumber + 2
```

Is `BeefyNextAuthoritySet.ID` the id after `Commitment.ValidatorSetID` (the handover case), or the id expected in the next commitment (the non-handover case)? It looks like these cases interpret `BeefyNextAuthoritySet.ID` differently.

Reply:
There are 2 cases. Handover with `currentValidatorSet +1 = Commitment.ValidatorSetID = MMRProof.Leaf.BeefyNextAuthoritySet.ID -1` and  non-handover with `(currentValidatorSet | currentValidatorSet + 1) = Commitment.ValidatorSetID = MMRProof.Leaf.BeefyNextAuthoritySet.ID`